test/mpi/threads/rma/mt_compareandswap.c

```diff
@@ -0,0 +1,195 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2018 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ *  Portions of this code were written by Intel Corporation.
+ *  Copyright (C) 2011-2018 Intel Corporation.  Intel provides this material
+ *  to Argonne National Laboratory subject to Software Grant and Corporate
+ *  Contributor License Agreement dated February 8, 2012.
+ */
+
+/* Multi-thread One-Sided MPI Compare and Swap Test
+ *
+ * This code performs compare_and_swap operations using multiple threads.
+ * Self communication, neighbor communication, contention cases are tested.
+ */
```

Comment:
Can we also add these communication patterns for all RMA operations/window creation types ?